### ISSUE TITLE
[quick_actions] try out pigeon Swift

### ISSUE THREAD

**Issue by hellohuanlin** [flutter/flutter#117112]
## Proposal

We recently added [Swift support for pigeon](https://github.com/flutter/packages/pull/976). It would be helpful to try it out and see if it works, if any potential issues, and gather feedbacks, etc. 

Note: `quick_actions` is fully migrated to Swift [here](https://github.com/flutter/flutter/issues/108750).

**Comment by hellohuanlin** [comment]
Hi @stuartmorgan, I saw your PR today to convert shared preference to use Swift pigeon on mac. Do you still think it's useful to try out Swift pigeon in iOS? Or do you think this should be lower priority?

**Comment by stuartmorgan** [comment]
Early on in Pigeon development, every new person that used it and every new plugin we applied it to found new issues and/or areas for improvement. So far I believe only I have used the Swift generator for 1P, and I have very limited experience with Swift so am unlikely to notice things that are unidiomatic.

So I think it would be great to have you do a conversation to get another set of eyes on the experience and file any issues you see, before we take the Swift generator out of experimental. (And my hope is that it would only be a day of work or so.) And there's an added benefit that it gets more people within the Flutter team familiar with using Pigeon, which I would like to fully switch our 1P development over to (see the new https://github.com/flutter/flutter/issues/117844). While you had this plugin fresh in your mind from the Swift rewrite seemed like an ideal time for that.

In the grand scheme of things though the priority of the conversion itself is low; as with the `quick_actions` rewrite it's the overall context that it helps move us toward that is the real benefit. But you and @jmagman are in a much better position than me to decide where that fits in overall iOS work priority. If you want to defer this, i certainly understand.
